services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -148,6 +149,22 @@ open class HankeServiceImpl(
         return savedHanke
     }
 
+    /**
+     * Create application when no existing hanke. Autogenerates hanke and applies application to it.
+     */
+    @Transactional
+    override fun generateHankeWithApplication(
+        application: Application,
+        userId: String
+    ): HankeWithApplications {
```

Comment:
The return type is HankeWithApplications, but only the applications are ever used, and only first one of those.

Reply:
Yes in this case. But this is already used and can be used elsewhere also. The real issue here is that I think hanke should contain applications at the domain level. It does not and we end up with stuff like this.

Reply:
So, the service creates hanke and an application. I think it is good that it returns both, even though the controller does not need the hanke information in this case.

Not returning the created hanke creates functionality that is cumbersome to test / verify.

Reply:
Yes, maybe it's better to return both for testability.